### ISSUE TITLE
gitui doesn't honor pushInsteadOf settings

### ISSUE THREAD

**Issue by mkhl** [extrawurst/gitui#953]
**Describe the bug**
Pushing to a remote that has a https url ignores an associated pushInsteadOf ssh url and prompts for a username.

**To Reproduce**
1. Clone a repository via https, e.g. `git clone https://github.com/extrawurst/gitui.git`
2. Add a pushInsteadOf ssh url, e.e.g by adding this to your git config:
    ```
    [url "git@github.com:extrawurst/"]
        pushInsteadOf = "https://github.com/extrawurst/"
    ```
3. Commit anything
4. See gitui prompt for a username

**Expected behavior**
gitui uses the configured ssh url and pushes successfully instead of prompting for credentials

**Context (please complete the following information):**
 - OS/Distro + Version: Fedora 34
 - GitUI Version: `gitui 0.16.2` (installed with dnf)
 - Rust version: `rustc 1.55.0 (Fedora 1.55.0-1.fc34)`

**Additional context**
[Documentation for pushInsteadOf](https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtpushInsteadOf)

**Comment by extrawurst** [comment]
Hm weird this should actually be working correctly since https://github.com/libgit2/libgit2/pull/3048

**Comment by extrawurst** [comment]
https://github.com/libgit2/libgit2/pull/6101 Is merged now we need to get it into git2-rs upstream

**Comment by extrawurst** [comment]
libgit2 release `1.4.0` released this now: https://github.com/libgit2/libgit2/releases/tag/v1.4.0

**Comment by mkhl** [comment]
this is fixed in [v0.21.0](https://github.com/extrawurst/gitui/releases/tag/v0.21.0) :tada:
